classes/03class/exercises/c03-aws02/alanlima/README.md

```diff
@@ -0,0 +1,197 @@
+# c03-aws02
+
+## Command Execution Output
+
+- Command used to create the Application Load Balancer (ALB)
+
+    * **Create Load Balancer**
+    
+    ```bash
+    $ aws elbv2 create-load-balancer \
+        --name devopsacademy-lb \
+        --scheme internet-facing \
+        --subnets subnet-0f82d6e49b9a6dafd subnet-0299eb6eb289cef54 \
+        --security-groups sg-098dda5e94f8a3855 \
+        --tags Key=DevopsClass,Value=c03 \
+        --type application \
+        --ip-address-type ipv4
+    ```
+
+    ```yaml
+    LoadBalancers:
+        - AvailabilityZones:
+        - LoadBalancerAddresses: []
+            SubnetId: subnet-0299eb6eb289cef54
+            ZoneName: ap-southeast-2b
+        - LoadBalancerAddresses: []
+            SubnetId: subnet-0f82d6e49b9a6dafd
+            ZoneName: ap-southeast-2a
+        CanonicalHostedZoneId: Z1GM3OXH4ZPM65
+        CreatedTime: '2020-07-09T06:06:10.500000+00:00'
+        DNSName: devopsacademy-lb-1691285263.ap-southeast-2.elb.amazonaws.com
+        IpAddressType: ipv4
+        LoadBalancerArn: arn:aws:elasticloadbalancing:ap-southeast-2:097922957316:loadbalancer/app/devopsacademy-lb/8514943c4cd7b139
+        LoadBalancerName: devopsacademy-lb
+        Scheme: internet-facing
+        SecurityGroups:
+        - sg-098dda5e94f8a3855
+        State:
+            Code: provisioning
+        Type: application
+    ```
+    
+    * **Create Target Group**
+    
+    ```bash
+    $ aws elbv2 create-target-group \
+        --name devopsacademy-lb-tg \
+        --protocol HTTP \
+        --port 80 \
+        --target-type instance \
+        --vpc-id vpc-027019faaf6756407
+    ```
+    
+    ```yaml
+    TargetGroups:
+    - HealthCheckEnabled: true
+      HealthCheckIntervalSeconds: 30
+      HealthCheckPath: /
+      HealthCheckPort: traffic-port
+      HealthCheckProtocol: HTTP
+      HealthCheckTimeoutSeconds: 5
+      HealthyThresholdCount: 5
+      LoadBalancerArns:
+      - arn:aws:elasticloadbalancing:ap-southeast-2:097922957316:loadbalancer/app/devopsacademy-lb/8514943c4cd7b139
+      Matcher:
+        HttpCode: '200'
+      Port: 80
+      Protocol: HTTP
+      TargetGroupArn: arn:aws:elasticloadbalancing:ap-southeast-2:097922957316:targetgroup/devopsacademy-lb-tg/f53590c0090c6267
+      TargetGroupName: devopsacademy-lb-tg
+      TargetType: instance
+      UnhealthyThresholdCount: 2
+      VpcId: vpc-027019faaf6756407
+    ```
+    
+    * **Register Targets / Create Listener** 
+    
+    ```bash
+    $ aws elbv2 register-targets \
+        --target-group-arn arn:aws:elasticloadbalancing:ap-southeast-2:097922957316:targetgroup/devopsacademy-lb-tg/f53590c0090c6267 \
+        --targets Id=i-061d30accff45a119 Id=i-04be230f0c0cd2fb2
```

Comment:
What happens if one of your instances goes down?

Probably the ASG will scale up another one to reach the desired state. But will it be registered and reachable via 
load balancer? (No)

For auto (de)registering instances created by the ASG you need to associate the ASG with the Target group:

```
aws autoscaling attach-load-balancer-target-groups \
    --auto-scaling-group-name devops-academy-asg \
    --target-group-arns arn:aws:elasticloadbalancing:ap-southeast-2:149613515908:targetgroup/devops-academy-tg/16871e3110304536
```

Reply:
Wow, that is a good point. So should I use attach to TG instead of `register-targets`?

Thanks for that, just learn a new command 🎊🎊

Reply:
I just tried here, it doesn't need to use `register-targets`.
Do you know any use-case to use `register-targets` instead of attach LB to TG?